af/analysis/analysis.py

```diff
@@ -0,0 +1,879 @@
+
+# Global db connectors
+dbengine = None
+conn = None
+
+# Send metrics to StatsD on docker host
+metrics = setup_metrics(host="172.17.0.1", name="analysis")
+
+
+def setup_database_connections():
+    global conn, dbengine
+    conn = psycopg2.connect(
+        dbname=dbname, user=dbuser, host=dbhost, password=dbpassword
+    )
+
+    dbengine = create_engine("postgresql+psycopg2://", creator=lambda: conn)
+
+
+def gen_table(name, df, cmap="RdYlGn"):
+    if cmap is None:
+        tb = df.style
+    else:
+        tb = df.style.background_gradient(cmap=cmap)
+    # df.style.bar(subset=['A', 'B'], align='mid', color=['#d65f5f', '#5fba7d'])
+    html = tb.render()
+    fn = os.path.join(output_directory, name + ".html")
+    print("Rendering", fn)
+    with open(fn, "w") as f:
+        f.write(html)
+
+
+def save(name, plt):
+    fn = os.path.join(output_directory, name + ".png")
+    print("Rendering", fn)
+    plt.get_figure().savefig(fn)
+
+
+def gen_plot(name, df, *a, **kw):
+    plt = df.plot(*a, **kw)
+    save(name, plt)
+
+
+def heatmap(name, *a, **kw):
+    fn = os.path.join(output_directory, name + ".png")
+    print("Rendering", fn)
+    h = sns.heatmap(*a, **kw)
+    h.get_figure().savefig(fn)
+
+
+def insert_into(tablename, q):
+    assert tablename in ("confirmed_stats", "confirmed_stats_asn")
+    # TODO: autoreconnect
+    with metrics.timer("insert_into." + tablename):
+        dbengine.execute(q)
+
+
+def query(q):
+    # TODO: add a label to generate metrics
+    print(" ".join(q.replace("\n", " ").split())[:300], "...")
+    # TODO: autoreconnect
+    with metrics.timer("query.unnamed"):
+        r = pd.read_sql_query(q, conn)
+    return r
+
+
+@metrics.timer("populate_interesting_inputs")
+def populate_interesting_inputs():
+    ## Used only once to create a persistent list of targets
+    dbengine.execute(
+        """
+        CREATE UNLOGGED TABLE interesting_inputs (
+            test_name  ootest NOT NULL ,
+            input text NOT NULL,
+            weight integer DEFAULT 1,
+            PRIMARY KEY (test_name, input)
+        )"""
+    )
+    insert_into(
+        "interesting_inputs",
+        """
+        INSERT INTO interesting_inputs
+        SELECT
+            test_name,
+            input,
+        COUNT(*) as cnt
+        FROM measurement
+        JOIN input ON input.input_no = measurement.input_no
+        JOIN report ON report.report_no = measurement.report_no
+        WHERE measurement_start_time > current_date - interval '2 days'
+        GROUP BY test_name, input
+    """,
+    )
+
+
+@metrics.timer("populate_countries")
+def populate_countries():
+    ## Used only once to create a persistent list of countries
+    dbengine.execute(
+        """
+        CREATE UNLOGGED TABLE countries (
+            probe_cc CHARACTER(2) NOT NULL,
+            msm_count BIGINT NOT NULL
+        );
+        CREATE INDEX ON countries (msm_count);
```

Comment:
Why are you creating an index on the count of measurements for the country?

You may also want to look at the `ooexpl_bucket_msm_count` table which has similar information in it already.